plugins/registry.js

```diff
@@ -579,10 +587,29 @@ export default class PluginRegistry {
                 pluginId: this.id,
                 component,
                 title,
+                icons: icons.map((icon) => ({...icon, icon: resolveReactElement(icon.icon)})),
```

Comment:
why pass them here too if they are processed in the action below `RECEIVED_PLUGIN_RHS_ICONS`?

Reply:
The action below is for subsequent updates, I only return the action below, not invoke it.

I could do something like:

```
        store.dispatch({
            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
            name: 'RightHandSidebarComponent',
            data: {
                id,
                pluginId: this.id,
                component,
                title,
                icons: [], // set empty array
            },
        });

        const setRHSPluginIcons = (newIcons = []) => {
            return (dispatch) => {
                dispatch({
                    type: ActionTypes.RECEIVED_PLUGIN_RHS_ICONS,
                    data: {
                        id,
                        icons: newIcons.map((icon) => ({...icon, icon: resolveReactElement(icon.icon)})),
                    },
                });
            };
        };

	store.dispatch(setRHSPluginsIcons(icons));
```